src/pylibsshext/session.pyx

```diff
@@ -383,6 +409,39 @@ cdef class Session(object):
         if rc == libssh.SSH_AUTH_ERROR or rc == libssh.SSH_AUTH_DENIED:
             raise LibsshSessionException("Failed to authenticate with password: %s" % self._get_session_error_str())
 
+    def authenticate_interactive(self, password):
+        """Authenticate this session using keyboard-interactive authentication.
+
+        :param password: The password to authenticate with.
+        :type password: str
+
+        :raises LibsshSessionException: If authentication failed.
+
+        :return: Nothing.
+        :rtype: NoneType
+        """
+        cdef int rc
+        cdef char should_echo
+        rc = libssh.ssh_userauth_kbdint(self._libssh_session, NULL, NULL)
+
+        while rc == libssh.SSH_AUTH_INFO:
+            prompt_count = libssh.ssh_userauth_kbdint_getnprompts(self._libssh_session)
+            if prompt_count > 0:
+                for prompt in range(prompt_count):
+                    prompt_text = libssh.ssh_userauth_kbdint_getprompt(self._libssh_session, prompt, &should_echo)
+                    if re.match(br"Password:\s*", prompt_text, flags=re.IGNORECASE):
```

Comment:
wouldn't it be better/easier/more efficient to avoid using regex?
```suggestion
                    if b"password:" == prompt_text.lower()[:9]:
```